jdbc-h2/src/main/java/io/kestra/runner/h2/H2WorkerJobQueue.java

```diff
@@ -4,27 +4,76 @@
 import io.micronaut.inject.qualifiers.Qualifiers;
+import lombok.extern.slf4j.Slf4j;
 
+import java.net.UnknownHostException;
 import java.util.function.Consumer;
 
+@Slf4j
 public class H2WorkerJobQueue implements WorkerJobQueueInterface {
-    QueueInterface<WorkerJob> workerTaskQueue;
+    JdbcQueue<WorkerJob> workerTaskQueue;
+    JdbcHeartbeat jdbcHeartbeat;
+    AbstractJdbcWorkerJobRunningRepository jdbcWorkerJobRunningRepository;
 
     @SuppressWarnings("unchecked")
     public H2WorkerJobQueue(ApplicationContext applicationContext) {
-        this.workerTaskQueue = (QueueInterface<WorkerJob>) applicationContext.getBean(
+        this.workerTaskQueue = (JdbcQueue<WorkerJob>) applicationContext.getBean(
             QueueInterface.class,
             Qualifiers.byName(QueueFactoryInterface.WORKERJOB_NAMED)
         );
+        this.jdbcHeartbeat = applicationContext.getBean(JdbcHeartbeat.class);
+        this.jdbcWorkerJobRunningRepository = applicationContext.getBean(AbstractJdbcWorkerJobRunningRepository.class);
     }
 
     @Override
     public Runnable receive(String consumerGroup, Class<?> queueType, Consumer<Either<WorkerJob, DeserializationException>> consumer) {
-        return workerTaskQueue.receive(consumerGroup, queueType, consumer);
+        return workerTaskQueue.receiveTransaction(consumerGroup, queueType, (dslContext, eithers) -> {
+
+            eithers.forEach(either -> {
+                if (either.isRight()) {
+                    log.error("Unable to deserialize an execution: {}", either.getRight().getMessage());
+                    return;
+                }
+                WorkerJob workerJob = either.getLeft();
+                WorkerInstance workerInstance = null;
+                try {
+                    workerInstance = jdbcHeartbeat.get().toWorkerInstance();
+                } catch (UnknownHostException e) {
+                    throw new RuntimeException(e);
+                }
+                WorkerJobRunning workerJobRunning;
+
+                if (workerJob instanceof WorkerTask workerTask) {
+                    workerJobRunning = WorkerTaskRunning.of(
+                        workerTask,
+                        workerInstance,
+                        0
+                    );
+                } else if (workerJob instanceof WorkerTrigger workerTrigger) {
+                    workerJobRunning = WorkerTriggerRunning.of(
+                        workerTrigger,
+                        workerInstance,
+                        0
+                    );
+                }
+                else {
+                    throw new IllegalArgumentException("Message is of type " + workerJob.getClass() + " which should never occurs");
+                }
+
+                jdbcWorkerJobRunningRepository.save(workerJobRunning, dslContext);
+
+                log.trace("Sending a workerJobRunning: {}", workerJobRunning);
+            });
+
+            eithers.forEach(consumer);
+        });
```

Comment:
this is duplicated in all components.
i wonder if inserting the worker running job inside the JdbcWorker would not be better.
Here, if someone else is reading this queue (which is a legitimate use case), it will duplicate the message.